Tests/FluentAssertions.Specs/Specialized/TaskAssertionSpecs.cs

```diff
@@ -180,6 +184,274 @@ public async Task When_task_completes_late_it_should_succeed()
+            timer.Complete();
+
+            // Assert
+            await action.Should().ThrowAsync<XunitException>().WithMessage(
+                "Expected a <System.InvalidOperationException> to be thrown, but no exception was thrown.");
+        }
+    }
+
+    public class ThrowWithinAsync
+    {
+        [Fact]
+        public async Task When_subject_is_null_it_should_throw()
+        {
+            // Arrange
+            Func<Task> action = null;
+
+            // Act
+            Func<Task> testAction = () => action.Should().ThrowWithinAsync<InvalidOperationException>(
+                100.Milliseconds(), "because we want to test the failure {0}", "message");
+
+            // Assert
+            await testAction.Should().ThrowAsync<XunitException>()
+                .WithMessage("*because we want to test the failure message*found <null>*");
+        }
+
+        [Fact]
+        public async Task When_subject_is_null_in_assertion_scope_it_should_throw()
+        {
+            // Arrange
+            Func<Task> action = null;
+
+            // Act
+            Func<Task> testAction = async () =>
+            {
+                using var _ = new AssertionScope();
+                await action.Should().ThrowWithinAsync<InvalidOperationException>(
+                    100.Milliseconds(), "because we want to test the failure {0}", "message");
+            };
+
+            // Assert
+            await testAction.Should().ThrowAsync<XunitException>()
+                .WithMessage("*because we want to test the failure message*found <null>*");
+        }
+
+        [Fact]
+        public async Task When_task_throws_fast_it_should_succeed()
+        {
+            // Arrange
+            var timer = new FakeClock();
+            var taskFactory = new TaskCompletionSource<bool>();
+
+            // Act
+            Func<Task> action = () =>
+            {
+                return taskFactory
+                    .Awaiting(t => (Task)t.Task)
+                    .Should(timer).ThrowWithinAsync<InvalidOperationException>(100.Milliseconds());
+            };
+            taskFactory.SetException(new InvalidOperationException("foo"));
+
+            // Assert
+            await action.Should().CompleteWithinAsync(1.Seconds());
+        }
+
+        [Fact]
+        public async Task When_task_throws_asynchronous_it_should_succeed()
+        {
+            // Arrange
+            var timer = new FakeClock();
+            var taskFactory = new TaskCompletionSource<bool>();
+
+            // Act
+            Func<Task> action = () =>
+            {
+                return Awaiting(() => (Task)taskFactory.Task)
+                    .Should(timer).ThrowWithinAsync<InvalidOperationException>(1.Seconds());
+            };
+            var _ = action.Invoke();
+            taskFactory.SetException(new InvalidOperationException("foo"));
+
+            // Assert
+            await action.Should().CompleteWithinAsync(1.Seconds());
+        }
```

Comment:
Why do we begin executing the task here?
`var _ = action.Invoke();`

That seems odd to me, as if we're not actually testing what we think we're testing.

Reply:
The test wants to validate the scenario, that the asynchronous method throws AFTER the synchronous portion has been completed. Therefore I need to "invoke".

Reply:
Roger.
From CC I see that this also exercises the `return exception;` statement on line 254.

I would have thought we could simplify this scenario with the following, but it fails when for me running the build script.
```cs
var action = async () =>
{
    await Task.Yield();
    throw new InvalidOperationException("foo");
};

// Act
Func<Task> testAction = () => action.Should().ThrowWithinAsync<InvalidOperationException>(1.Seconds());

// Assert
await testAction.Should().NotThrowAsync();
```